packet.go

```diff
@@ -290,6 +290,9 @@ func recvPacket(r io.Reader, alloc *allocator, orderID uint32) (uint8, []byte, e
 		b = make([]byte, length)
 	}
 	if _, err := io.ReadFull(r, b[:length]); err != nil {
+		if err == io.EOF {
+			err = io.ErrUnexpectedEOF
```

Comment:
I would like to document why we’re changing the `err` here. “ReadFull will only return EOF if it has read no bytes. In this case, that means a partial packet, and thus unexpected.”

Reply:
Done.